src/completion/completeSnippets.ts

```diff
@@ -53,11 +53,13 @@ function doesStateSupportErrorHandling(node: ASTNode): boolean {
     return ERROR_HANDLING_STATES.includes(typeNode?.valueNode?.value as string)
 }
 
-export default function completeSnippets(node: ASTNode | undefined): CompletionItem[] {
+export default function completeSnippets(node: ASTNode | undefined, offset: number, shouldShowStateSnippets?: boolean): CompletionItem[] {
     if (node) {
-        if (isChildOfStates(node)) {
+        // If the value of shouldShowStateSnippets is false prevent the snippets from being displayed
+        if ((isChildOfStates(node) && shouldShowStateSnippets !== false) || shouldShowStateSnippets) {
```

Comment:
Can this be simplified to (shouldShowStateSnippets === undefined ? (isChildOfStates(node) : shouldShowStateSnippets)